docs/docs-content/clusters/edge/trusted-boot/generate-key.md

```diff
@@ -0,0 +1,90 @@
+---
+sidebar_label: "Generate Keys for Trusted Boot"
+title: "Generate Keys for Trusted Boot"
+description: "Learn about how to generate keys for Trusted Boot."
+hide_table_of_contents: false
+sidebar_position: 10
+tags: ["edge"]
+---
+
+Trusted Boot works by signing the Edge Installer image and provider images with keys, and only allowing images signed
+with the trusted keys to operate during the Edge host boot process. This page guides you through the process of
+generating keys to be used in building Edge artifacts.
+
+The key generation process produces three pairs of keys and a platform Configuration Register (PCR) policy private key,
+and each pair of keys fulfill different purposes. The following table provides a brief overview of which keys are used
+in which Trusted Boot EdgeForge and deployment process.
+
+| Keys                     | Key Generation | Build Installer ISO | Building Provider Images | Installation |
+| ------------------------ | -------------- | ------------------- | ------------------------ | ------------ |
+| PK & KEK (private)       | ✅             | Not needed          | Not needed               | Not needed   |
+| PK & KEK (public)        |                | ✅                  | Not needed               | ✅           |
+| DB (public)              |                | ✅                  | ✅                       | Not needed   |
+| DB (private)             |                | ✅                  | ✅                       | Not needed   |
+| PCR policy Key (private) |                | ✅                  | Not needed               | ✅           |
+
+## Prerequisites
+
+:::warning
+
+All security provided by Trusted Boot assumes that the private keys are kept secure. We suggest that you perform key
+generation in an airgapped environment and move the private keys to a secure location immediately after generating them.
+
+:::
+
+- A physical or virtual Linux machine with _AMD64_ (also known as x86_64) processor architecture to build the Edge
+  artifacts. You can issue the following command in the terminal to check your processor architecture.
+
+  ```bash
+  uname -m
+  ```
+
+- Minimum hardware configuration of the Linux machine:
+
+  - 4 CPU
+  - 8 GB memory
```

Comment:
I believe there may be new memory (RAM) requirements for building, check with @nianyush

Reply:
32G will be minimum for builder

Reply:
This document only talks about generating keys. Since we suggest users do key generations on a different machine than they build the ISO and provider images on, is this requirement for key generation as well?